aeppl/mixture.py

```diff
@@ -238,7 +240,7 @@ def get_stack_mixture_vars(
     return mixture_rvs, join_axis
 
 
-@local_optimizer(subtensor_ops)
+@local_optimizer(subtensor_ops + (Elemwise,))
```

Comment:
A distinct `local_optimizer` is probably the best approach here.  

Also, if the `Op` class being tracked has only one type instance, that instance can be tracked directly, which makes the checking (i.e. `dict` hashing and `Op.__eq__`) of `Elemwise` `Op`s more accurate.  Specifically, the tracking functionality will use the `Elemwise.scalar_op` value, as well, so it won't need to be checked manually here.

Reply:
What is this tracking functionality that you mention? I am not sure to fully understand the idea that you're conveying here. In the separate `local_optimizer` for `Elemwise`, I am still checking that the `scalar_op` is indeed a `Switch`. Is this still necessary?

Reply:
The first argument to `local_optimizer` is the list of `Op` instances or classes to which the rewrite is applied.  Having separate `local_optimizer`s for distinct `Op`s reduced the need for redundant/manual filtering within the rewrite.

Also, tracking something like `at.add` (i.e. an `Op` instance) will make sure that the rewrite sees only nodes using that `Op` instance, where as tracking `Elemwise` classes will require an `isinstance(op.scalar_op, aesara.scalar.basic.Add)` check within the rewrite.